tests/high_frequency_telemetry/test_high_frequency_telemetry.py

```diff
@@ -0,0 +1,740 @@
+
+    The test validates that the actual Msg/s values are within an acceptable range
+    of the expected frequency based on the configured poll interval.
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    profile_name = f"poll_interval_profile_{poll_interval_us}"
+    group_name = "PORT"
+
+    # Get available ports from topology (try to get 2 ports, minimum 1 required)
+    test_ports = get_available_ports(duthost, tbinfo, desired_ports=2, min_ports=1)
+
+    logger.info(f"Testing poll interval: {poll_interval_us} μs (expected Msg/s: {expected_msg_per_sec})")
+    logger.info(f"Using ports for testing: {test_ports}")
+
+    try:
+        # Step 1: Set up high frequency telemetry profile with specific poll interval
+        setup_hft_profile(
+            duthost=duthost,
+            profile_name=profile_name,
+            poll_interval=poll_interval_us,
+            stream_state="enabled"
+        )
+
+        # Step 2: Configure port group with specific ports and counters
+        setup_hft_group(
+            duthost=duthost,
+            profile_name=profile_name,
+            group_name=group_name,
+            object_names=test_ports,
+            object_counters=["IF_IN_OCTETS"]
+        )
+
+        logger.info(f"HFT group configuration completed for {poll_interval_us} μs poll interval")
+
+        # Verify the configuration is actually applied by checking Redis
+        logger.info("Verifying HFT configuration in Redis...")
+        verify_cmd = "redis-cli -n 4 HGETALL 'HFT_PROFILE|" + profile_name + "'"
+        config_result = duthost.shell(verify_cmd, module_ignore_errors=True)
+        if config_result['rc'] == 0 and config_result['stdout']:
+            logger.info(f"HFT profile configuration: {config_result['stdout']}")
+        else:
+            logger.warning(f"Could not verify HFT profile configuration: {config_result}")
+
+        verify_group_cmd = "redis-cli -n 4 HGETALL 'HFT_GROUP|" + profile_name + "|" + group_name + "'"
+        group_result = duthost.shell(verify_group_cmd, module_ignore_errors=True)
+        if group_result['rc'] == 0 and group_result['stdout']:
+            logger.info(f"HFT group configuration: {group_result['stdout']}")
+        else:
+            logger.warning(f"Could not verify HFT group configuration: {group_result}")
+
+        # Give some time for the configuration to be applied
+        logger.info("Waiting 10 seconds for HFT configuration to take effect...")
+        time.sleep(10)
+
+        # Step 3: Run countersyncd and capture output for sufficient time to get stable measurements
+        # Use longer timeout for slower poll intervals to ensure we get enough samples
+        timeout = max(120, int(10 / expected_msg_per_sec) + 60) if expected_msg_per_sec > 0 else 180
+        logger.info(f"Running countersyncd for {timeout} seconds to capture stable measurements")
+
+        result = run_countersyncd_and_capture_output(duthost, timeout=timeout)
```

Comment:
As we discussed on Teams, on slower poll intervals you also need to increase the reporting interval to get the correct Msg/s